Part 1/X: Finishing Zesko Virology Rebalancing Some Items

DatBoiTim opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
I had talked with Zesko regarding Virology. They asked if I was willing to do some rebalancing and such to get viro to where they wanted it.

This Entailed:

- Lower Extrapolator Cooldown
- Higher Virus Crate Price(3750)
- All potential Maint Disease Things have a minimum of 75% chance to have a disease(varies by object)
- All Maint Diseases have a minimum of three symptoms
- Maint diseases won't spawn anything below a level 3 symptom (most items generate level 4 minimum)
- Vector's Disease Generation was buffed, but it is slightly weaker than maint(Vector is a freebie after all)

Now, while this should buff maints, I can't tell for sure if it's far enough.

I also added 2 new virology crates:

Extrapolator Replacements
Pandemic Replacement

So while rushing Cargo for virology may not be the best answer. Cargo can ~~do virology on their own~~ replace virological equipment when lost or destroyed.


## TODO

- [x] Buff Random Viruses
- [x] Buff Maint Virus Spawns

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Adjusts Virology Gameplay to Make Cargo Rush Less Good, While Also Allowing Cargo to have a role in replacement ~~or doing it themself~~
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: PANDEMIC Replacement Crate
add: Extrapolator replacement Crate
tweak: Decreased Virus Extrapolator Cooldown
tweak: Increased Virus Sample Crate Cost
tweak: Increased Maint Virus Spawn Rate
tweak: Makes Maint Viruses Better
code: Added a minimum symptom level argument used in random disease generation
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

DatBoiTim committed:
Adjusts Viro Gameplay

DatBoiTim committed:
Wrong Virus Crate New Extrapolator Crate

DatBoiTim committed:
Pandemic Replacement Crate

DatBoiTim committed:
Adjusts Virology Crate Costs

DatBoiTim committed:
Finishing Up PANDEMIC Crate Define

zeskorion commented:
my rationale is essentially that viro gameplay goes as follows:
get med budget
get a t3 scanner
rush cargo and get like 6000 bucks worth of sample diseases and have every symptom
done

this is (going to) make intended gameplay, being maint exploration to find stuff and getting out da lab, actually worth it

leestriter commented:
I mean as is, getting every symptom takes a lot of crates. I've many times spent the entire medical budget and not gonna what I need, and very often the CMO doesnt even give me the 1500 for one crate. I think wed need a big buff to random viruses to justify this 

Yeah I can say right now, no CMO is ever going to give a viro 5 k. And that's for a single crate. Unless we see a huge buff to random spawns, this may as well be a removal of level 9 symptoms, and the removal of level 7 and 8 when there are no miners. Also, why let cargo just buy anything? What happened to making stuff in science? I'm pretty sure sci can make new pandemic boards. And you've left the equipment there for anyone to start their own viro lab, an assistant can start their own virology lab. Also, viro should be able to open a virus crate not just cmo. 

Could you say the exact changes publically instead of just in code? People should see the exact pricing without looking at code. Also, don't list things in the change log that aren't yet parts of the PR, it is deceptive. When/if those things are added put them in the changelog. 

My recommendation? A pandemic or extrapolator from cargo should be very expensive, only for emergencies. A virus crate should be raised to 3000 not 5000, so its expensive enough you wont just spam them but still possible if you do bounties.

Making maint diving worth it is a good idea, but letting cargo make their own virology and making samples so expensive that they are effectively removed is not the way to do it. If you properly fix maint viruses, you shouldn't need to change the price of cargo crates at all.

As for zesko's response, cmo is not gonna give you the entire med budget like ever (I normally get at most 1500, normally nothing). With RNG even 6000 is sometimes not enough in the current system. Mining/science don't always rush high tier parts and sometimes you will be using the base extrapolator.

leestriter commented:
I have seriously edited my original message.

I wanna make it clear I have nothing against datboitim or zesko. I think this pr has a great purpose, and I know its work in progress, as a viro main I was just letting you know the things I thought were problematic and why I thought they should be fixed. I think this will end up an amazing pr. Datboitim is a valuable member of the community and I am very glad to see Zesko involved again.

zeskorion commented:
.

Random viruses themselves cant really be buffed, but maint viruses can be buffed to be nearly equivalent. Getting high tier parts researchdd isnt that difficult and i've done it myself at times, though

I used to have the CMO just hand me the budget sometimes, or give me larger amounts of money. You can also trade viro bounties for sample crates( if cost for a crate here is lowered to 4000 you can get one every round. You can ususlly find shit to satisffy a bounty in maints)

leestriter commented:
> .
> 
> Random viruses themselves cant really be buffed, but maint viruses can be buffed to be nearly equivalent. Getting high tier parts researchdd isnt that difficult and i've done it myself at times, though
> 
> I used to have the CMO just hand me the budget sometimes, or give me larger amounts of money. You can also trade viro bounties for sample crates( if cost for a crate here is lowered to 4000 you can get one every round. You can ususlly find shit to satisffy a bounty in maints)

In my experience the CMO will not give me the budget or even a large portion of it, and from discussions with other viros it doesn't happen for them either. A lot has changed since you last played, the rules for heads are more strict and that could be a part.

What if we split the difference and go 3500? And what's the average value of a viro bounty?

DatBoiTim commented:
> .
> 
> 
> 
> Random viruses themselves cant really be buffed, but maint viruses can be buffed to be nearly equivalent. Getting high tier parts researchdd isnt that difficult and i've done it myself at times, though
> 
> 
> 
> I used to have the CMO just hand me the budget sometimes, or give me larger amounts of money. You can also trade viro bounties for sample crates( if cost for a crate here is lowered to 4000 you can get one every round. You can ususlly find shit to satisffy a bounty in maints)

Yeah I'll probably nudge price a bit. I am not removing the other stuff since I do want cargo to A) Be an option for replacing broken equipment, B) To potentially make their own lab in cases of antags

leestriter commented:
> > .
> > Random viruses themselves cant really be buffed, but maint viruses can be buffed to be nearly equivalent. Getting high tier parts researchdd isnt that difficult and i've done it myself at times, though
> > I used to have the CMO just hand me the budget sometimes, or give me larger amounts of money. You can also trade viro bounties for sample crates( if cost for a crate here is lowered to 4000 you can get one every round. You can ususlly find shit to satisffy a bounty in maints)
> 
> Yeah I'll probably nudge price a bit. I am not removing the other stuff since I do want cargo to A) Be an option for replacing broken equipment, B) To potentially make their own lab in cases of antags



> > .
> > Random viruses themselves cant really be buffed, but maint viruses can be buffed to be nearly equivalent. Getting high tier parts researchdd isnt that difficult and i've done it myself at times, though
> > I used to have the CMO just hand me the budget sometimes, or give me larger amounts of money. You can also trade viro bounties for sample crates( if cost for a crate here is lowered to 4000 you can get one every round. You can ususlly find shit to satisffy a bounty in maints)
> 
> Yeah I'll probably nudge price a bit. I am not removing the other stuff since I do want cargo to A) Be an option for replacing broken equipment, B) To potentially make their own lab in cases of antags

I'd just price that equipment to help make sure its only used for those two things, and not bought by randos. I do like having the equipment in cargo, just don't think it should be cheap. Cause it provides a hard counter to stealth diseases and opens up potential for grief.

DatBoiTim committed:
Adjusts Crate Costs

DatBoiTim committed:
Increases Odds For Common Maint Disease Sources To Contain Diseases

DatBoiTim committed:
Missed the Gibs

DatBoiTim commented:
Fucking linters

DatBoiTim commented:
Okay trying to re-run linters

DatBoiTim committed:
Buffs maint disease generation

DatBoiTim commented:
Not sure how many PRs will be created for this, but this is one of a few

DatBoiTim commented:
Goddamnit I didn't delete all of template CL.

leestriter commented:
I like what I'm seeing. Hard to judge some of the numbers till I see them in action, honestly.

zeskorion commented on code/game/objects/effects/decals/cleanable/misc.dm:
```diff
@@ -149,8 +149,8 @@
-	if(prob(90))//vomit is much more likely to be diseased than blood is
-		var/datum/disease/advance/R = new /datum/disease/advance/random(rand(2, 6), 6+(rand(1, 3)))
+	if(prob(95))//vomit is much more likely to be diseased than blood is
+		var/datum/disease/advance/R = new /datum/disease/advance/random(rand(3, 6), (rand(4, 9)))
```
you're actually nerfing symptom level here

zeskorion commented on code/modules/mob/living/simple_animal/friendly/hamster.dm:
```diff
@@ -45,7 +45,7 @@
 		message_admins("Vector was roundstart infected with [vector_disease.name]. Don't lynch the virologist!")
 		log_game("Vector was roundstart infected with [vector_disease.name].")
-	var/datum/disease/advance/R = new /datum/disease/advance/random(rand(1, 3))
+	var/datum/disease/advance/R = new /datum/disease/advance/random(rand(2, 5), rand(3,9))
```
ditto. if no symptom level number is given it caps at 9 by default

DatBoiTim replied:
Altered Disease Generation to have a minimum level. Changing disease generation for maints to match.

zeskorion commented on code/modules/mob/living/simple_animal/friendly/mouse.dm:
```diff
@@ -38,8 +38,8 @@
-	if(prob(40))
-		var/datum/disease/advance/R = new /datum/disease/advance/random(rand(2, 4))
+	if(prob(75))
+		var/datum/disease/advance/R = new /datum/disease/advance/random(rand(3, 6), rand(4, 9))
```
ditto

DatBoiTim replied:
Ah shit did I get the symptom count and symptom level backwards

zeskorion replied:
Yeah

DatBoiTim replied:
Should be fixed now.

DatBoiTim replied:
Actually I am changing up the random disease generation a tad

By Default it will have a minimum of level 1
But you can pass an arg to skew towards specific levels

DatBoiTim committed:
Fixes the random disease generation humans.dm

I had put the symptom count where symptom level was and symptom level where symptom count

DatBoiTim committed:
Fixes Mouse.dm Virus Generation

DatBoiTim committed:
Update syringes.dm

DatBoiTim committed:
Update hamster.dm

DatBoiTim committed:
Update misc.dm

DatBoiTim committed:
Disease Generation Changes

Minimum Level Arg added to Random Disease Generation
Changed Maint Disease Generation to Match Accordingly

ItsTisme commented:
I have to say aswel, replacing viro's pandamic is dead simple as its literly a computer frame and a board, with research aviable round start for like 2500/5000 points(cant remember the exact node). Not to mention how rare it would be for viro's machine to break in the first place. 

This will mainly just be used for cargo to make aids and sell the vaccine for 500 credits  a pop

DatBoiTim commented:
> I have to say aswel, replacing viro's pandamic is dead simple as its literly a computer frame and a board, with research aviable round start for like 2500/5000 points(cant remember the exact node). Not to mention how rare it would be for viro's machine to break in the first place.
> 
> This will mainly just be used for cargo to make aids and sell the vaccine for 500 credits a pop

Cargo is getting hammered out or being used as an excuse to nerf other departments in silly ways. I'd rather give cargo an excuse to be a fast track past making new machines than not.

utilbird commented on code/modules/mob/living/simple_animal/friendly/mouse.dm:
```diff
@@ -38,8 +38,8 @@
-	if(prob(40))
-		var/datum/disease/advance/R = new /datum/disease/advance/random(rand(2, 4))
+	if(prob(75))
+		var/datum/disease/advance/R = new /datum/disease/advance/random(rand(3, 6), 9, 1+rand(2,3))
```
```suggestion
		var/datum/disease/advance/R = new /datum/disease/advance/random(rand(3, 6), 9, rand(3,4))
```

utilbird commented on code/modules/reagents/reagent_containers/syringes.dm:
```diff
@@ -209,16 +209,16 @@
-	if(prob(50))
-		var/datum/disease/advance/R = new /datum/disease/advance/random(rand(2, 5), rand(6, 9))
+	if(prob(75))
+		var/datum/disease/advance/R = new /datum/disease/advance/random(rand(3, 6), 9, 1+rand(2,3))
```
```suggestion
		var/datum/disease/advance/R = new /datum/disease/advance/random(rand(3, 6), 9, rand(3,4))
```

zeskorion replied:
🙄

DatBoiTim committed:
Update code/modules/mob/living/simple_animal/friendly/mouse.dm

Co-authored-by: TheFakeElon <59686430+TheFakeElon@users.noreply.github.com>

DatBoiTim committed:
Update code/modules/reagents/reagent_containers/syringes.dm

Co-authored-by: TheFakeElon <59686430+TheFakeElon@users.noreply.github.com>
